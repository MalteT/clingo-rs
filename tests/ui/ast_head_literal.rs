use clingo::*;

fn main() {
    let sym = Symbol::create_id("test", true).unwrap();
    let term = ast::Term::from(sym);
    let lit = ast::Literal::from_term(ast::Sign::None, &term);
    let hlit = ast::HeadLiteral::from(&lit);
    drop(lit);
    let _end = hlit;
}