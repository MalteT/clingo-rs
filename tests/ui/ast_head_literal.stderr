error[E0505]: cannot move out of `lit` because it is borrowed
 --> $DIR/ast_head_literal.rs:8:10
  |
7 |     let hlit = ast::HeadLiteral::from(&lit);
  |                                       ---- borrow of `lit` occurs here
8 |     drop(lit);
  |          ^^^ move out of `lit` occurs here
9 |     let _end = hlit;
  |                ---- borrow later used here

For more information about this error, try `rustc --explain E0505`.
